### PR TITLE
[3.x] Increase VARIANT_ARG_MAX to 8

### PR DESCRIPTION
**Pull request by Rubonnek**
Fixes:

- #54185
- #26483

Aside from the issues above, the other reason for increasing `VARIANT_ARG_MAX` to 8 is because it was done on `master` already to fix similar issues.

See:

- #50206.

### CODE REVIEW

**Comment by Rubonnek** [comment]
Updated a code block in `AnimationTree` that also depends on `VARIANT_ARG_MAX`.

I should say that this pull request now also allows users to call `TileMap.set_cell` successfully through an `AnimationPlayer` and `AnimationTree`, given that the we are now able to pass the number of parameters required to properly call such functions.

File: editor/animation_track_editor.cpp [hunk 1]
```diff
@@ -556,7 +556,8 @@ class AnimationTrackKeyEdit : public Object {
 			} break;
 			case Animation::TYPE_METHOD: {
 				p_list->push_back(PropertyInfo(Variant::STRING, "name"));
-				p_list->push_back(PropertyInfo(Variant::INT, "arg_count", PROPERTY_HINT_RANGE, "0,5,1"));
+				static_assert(VARIANT_ARG_MAX == 8, "PROPERTY_HINT_RANGE needs to be updated if VARIANT_ARG_MAX != 8");
+				p_list->push_back(PropertyInfo(Variant::INT, "arg_count", PROPERTY_HINT_RANGE, "0,8,1"));
```
Comment (akien-mga) [inline editor/animation_track_editor.cpp:560]: We discussed this in a PR review meeting and we're not sure it's worth changing the whole `VARIANT_ARG_MAX` just for this - the limit on animation code doesn't seem related to `VARIANT_ARG_MAX` and could likely be lifted to allow any number of arguments, no?

Comment (Rubonnek) [reply editor/animation_track_editor.cpp:560]: In this code the user is modifying a key within a method-type track of an `AnimationPlayer`, right? The number of arguments for a method-type track in an `AnimationPlayer` is bound by the maximum number of arguments the `AnimationPlayer` itself supports when playing this track. `VARIANT_ARG_MAX` is related. I fail to see how it could not.  Keep an eye on `Animation::TYPE_METHOD`.

When the AnimationPlayer attempts to play the method-type track, it will run the following code (only pay attention to the first line, the last line, and `params` near the bottom):
https://github.com/godotengine/godot/blob/b197de6f5f1e61a9cecf0a29aadea0fb024b1edf/scene/animation/animation_player.cpp#L527-L548

The very last line of that code block has a direct reference to `VARIANT_ARG_MAX`. That check is there because further below `AnimationPlayer` reconstructs the method call extracted from the track:

https://github.com/godotengine/godot/blob/b197de6f5f1e61a9cecf0a29aadea0fb024b1edf/scene/animation/animation_player.cpp#L556-L573
The `params` array holds the number of arguments previously saved on the track through the `AnimationTrackEditor`. Everytime `VARIANT_ARG_MAX` is updated, that last code block needs to be updated to support the maximum number of arguments.

File: scene/animation/tween.cpp [hunk 2]
```diff
@@ -1520,8 +1534,13 @@ bool Tween::interpolate_deferred_callback(Object *p_object, real_t p_duration, S
 	data.delay = 0;
 
 	// Collect arguments for the callback
+	static_assert(VARIANT_ARG_MAX == 8, "This code needs to be updated if VARIANT_ARG_MAX != 8");
 	int args = 0;
-	if (p_arg5.get_type() != Variant::NIL) {
+	if (p_arg7.get_type() != Variant::NIL) {
```
Comment (akien-mga) [inline scene/animation/tween.cpp:1539]: Missing `p_arg8`.

Comment (Rubonnek) [reply scene/animation/tween.cpp]: Fixed,

**Review by akien-mga — APPROVED** [review]
Discussed in PR review meeting, we're slightly concerned about the potential performance impact but given the places where it's used, it _should_ be fine.

For Godot 4.0, this needs to be changed to use variadic templates so we can dehardcode things properly.

**Comment by akien-mga** [comment]
Thanks!
